src/nnbench/core.py

```diff
@@ -224,7 +239,8 @@ def decorator(fn: Callable) -> list[Benchmark]:
         benchmarks = []
         names = set()
         varnames = iterables.keys()
-        for values in itertools.product(*iterables.values()):
+        cartesian_product = list(itertools.product(*iterables.values()))
```

Comment:
`list` here forces execution of the generator. I think that is no problem as the number of benchmarks will not be too computationally extensive. Additionally, we nee the number of values in the cartesian product as well as the index anyway, so we might as well unravel the generator here instead of in the following for loop.